lib/assertions.js

```diff
@@ -766,39 +766,65 @@ module.exports = (expect) => {
     );
   });
 
-  expect.addAssertion('<array-like> [not] to contain <any+>', function (
-    expect,
-    subject
-  ) {
-    const args = Array.prototype.slice.call(arguments, 2);
-    expect.withError(
-      () => {
-        args.forEach((arg) => {
-          expect(
-            subject &&
-              Array.prototype.some.call(subject, (item) =>
-                expect.equal(item, arg)
-              ),
-            '[not] to be truthy'
-          );
-        });
-      },
-      (e) => {
-        expect.fail({
-          diff:
-            expect.flags.not &&
-            ((output, diff, inspect, equal) =>
-              diff(
-                subject,
-                Array.prototype.filter.call(
+  expect.addAssertion(
+    [
+      '<array-like> to [only] contain <any+>',
+      '<array-like> [not] to contain <any+>',
+    ],
+    function (expect, subject) {
+      const args = Array.prototype.slice.call(arguments, 2);
+
+      expect.withError(
+        () => {
+          args.forEach((arg) => {
+            expect(
+              subject &&
+                Array.prototype.some.call(subject, (item) =>
+                  expect.equal(item, arg)
+                ),
+              '[not] to be truthy'
+            );
+          });
+          if (expect.flags.only) {
+            expect(
+              subject &&
+                Array.prototype.every.call(subject, (item) =>
+                  Array.prototype.some.call(args, (arg) =>
+                    expect.equal(item, arg)
+                  )
+                ),
+              'to be truthy'
+            );
+          }
+        },
+        (e) => {
+          expect.fail({
+            diff(output, diff, inspect, equal) {
+              if (expect.flags.not) {
+                return diff(
                   subject,
-                  (item) => !args.some((arg) => equal(item, arg))
-                )
-              )),
-        });
-      }
-    );
-  });
+                  Array.prototype.filter.call(
+                    subject,
+                    (item) => !args.some((arg) => equal(item, arg))
+                  )
+                );
+              } else if (expect.flags.only) {
+                var excessItems = Array.prototype.filter.call(subject, (item) =>
+                  args.some((arg) => equal(item, arg))
+                );
+                var missingItems = Array.prototype.filter.call(
+                  args,
+                  (arg) => !subject.some((item) => equal(arg, item))
+                );
+
+                return diff(subject, excessItems.concat(missingItems));
```

Comment:
```suggestion
                return diff(subject, [...excessItems, ...missingItems]);
```

Reply:
Wasn't sure about the code-style in the project - tried to strike a balance based on the surrounding code 👯 . ES6 is all good?